std/string.d

```diff
@@ -3144,6 +3291,104 @@ if (isConvertibleToString!Range)
     });
 }
 
+/++
+    Strips both leading and trailing extra characters as specified in the
+    second argument.
+
+    Params:
+        str = string or random access range of characters
+        chars = list of chars to strip
+
+    Returns:
+        slice of $(D str) stripped of leading and trailing extra characters.
+
+    See_Also:
+        Generic stripping on ranges: $(REF _strip, std, algorithm, mutation)
+  +/
+auto strip(Range)(Range str, string chars)
+if (isSomeString!Range ||
+    isRandomAccessRange!Range && hasLength!Range && hasSlicing!Range &&
```

Comment:
ditto, we can relax the constraints here to just `isBidirectionalRange!Range`, and that should be sufficient to implement this function.